homeassistant/components/lock/kiwi.py

```diff
@@ -0,0 +1,100 @@
+"""
+Support for the KIWI.KI lock platform.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/lock.kiwi/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.lock import (LockDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_USERNAME, ATTR_ID, ATTR_LONGITUDE, ATTR_LATITUDE,
+    STATE_LOCKED, STATE_UNLOCKED)
+from homeassistant.helpers.event import async_call_later
+from homeassistant.core import callback
+
+REQUIREMENTS = ['kiwiki-client==0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_TYPE = 'hardware_type'
+ATTR_PERMISSION = 'permission'
+ATTR_CAN_INVITE = 'can_invite_others'
+
+UNLOCK_MAINTAIN_TIME = 5
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the KIWI lock platform."""
+    from kiwiki import KiwiClient
+    kiwi = KiwiClient(config[CONF_USERNAME], config[CONF_PASSWORD])
```

Comment:
I guess that there will be a traceback if the credentials are wrong. The setup should fails otherwise the users end up with a non-working platform.